### PR TITLE
Fix OAuth Social sample

### PR DESCRIPTION
**Pull request by HaoK**
Like cookies OAuthAuthenticationOptions now uses instances of options.

@Tratcher @PinpointTownes this fixes things, but things still feel a bit clunky.  I'd love to nuke ConfigureOptions entirely potentially, so then we could take over the lambda to be configuring a new instance of the options instead of tweaking the one that came from DI.  Thoughts?

### CODE REVIEW

**Comment by HaoK** [comment]
Fixes https://github.com/aspnet/Security/issues/431

**Comment by Tratcher** [comment]
Agreed, it never made sense to me to pull these from DI

**Comment by Tratcher** [comment]
:shipit: and fix the other extensions separately.

**Comment by HaoK** [comment]
5bc13cbd6b3d3fcc1dc430dcb8841983d3731896

**Comment by PinpointTownes** [comment]
My 2 cents:
* Removing ConfigureOptions is something I've been advocating for a long time, so yeah, nuke it in a future commit (specially since it's now unused, so there's no reason to keep it). 
* Being able to configure the options via DI was a pretty cool thing and was useful to configure per-environment settings. I'm not sure to understand why you don't want to provide an extension that uses options from DI. If you really want to remove it, I think it should be part of a larger work removing it from everywhere it is used (in MVC for instance). 
* Removing the delegate-style extension is clearly a regression from a code style perspective and was not necessary for this fix. Please consider bringing it back.

**Comment by HaoK** [comment]
There's a fix larger issues mixed into these:

- ConfigureOptions existed before mostly just to package both the option name and Action, but now that name is gone, it has no purpose, so that almost certainly will get nuked soonish.

- Well its certainly not going to be removed everywhere at this point, so its just a matter of what makes sense for multi instance middlewares.  Cookies does currently support both DI/instance via 2 overloads.  They should probably be consistent (I could go either way though)

- The main reason for nuking the delegate, was because it conflicts with the DI style overload (which didn't make the cut).  But if it comes back,I'd have lean in favor of the instance one taking the delegate parameters, and requiring UseXyz to take no parameters and just use what's in DI

**Comment by HaoK** [comment]
Discussed with @divega today and we decided that for now, we will go with only UseXyz(XyzOption), nuking all DI support for now for middlewares.  This lets us simplify and also remove bunch of overloads and complexity.  We can always add back sugar to make getting options from DI easier later

**Comment by PinpointTownes** [comment]
Understood. But please, bring back the delegate style extension.

**Comment by HaoK** [comment]
@divega can you summarize again why we decided against adding back the delegate style overload?  I know we agreed not to do it, but I can't recall exactly what our reasoning was for that part.

**Comment by PinpointTownes** [comment]
@divega ping! :package:

**Comment by divega** [comment]
@PinpointTownes sorry I didn't answer on this thread on time. We really wanted to reduce the number of patterns we used to pass options to middleware to the bare minimum that we actually needed. Anyway, as you already saw in https://github.com/aspnet/Security/pull/464, the lambda style overloads survived :smile:
